src/jobs/deploy_latest_codacy_chart.yaml

```diff
@@ -0,0 +1,92 @@
+description: "Set artifact version and helm repository on requirements.yaml"
+
+parameters:
+  release_name:
+    description: "Helm release name"
+    type: string
+    default: "codacy-unstable"
+  helm_channel:
+    description: "Helm channel from which to get the artifact"
+    type: string
+    default: "unstable"
+  namespace:
+    description: "Namespace for the codacy installation"
+    type: string
+    default: "codacy-unstable"
+  set_kubeconfig_command:
+    description: "Command (shell) to set the kubeconfig context onto which codacy will be installed"
+    type: string
+    default: ""
+  codacy_url:
+    description: "Codacy url for git hooks"
+    type: string
+    default: "http://k8s.unstable.dev.codacy.org"
+
+executor: aws
+
+environment:
+  RELEASE_NAME: << parameters.release_name >>
+  NAMESPACE: << parameters.namespace >>
+  SET_KUBECONFIG_COMMAND: <<parameters.set_kubeconfig_command>>
+  CHANNEL: << parameters.helm_channel >>
+  HELM_REPO: https://charts.codacy.com/<< parameters.helm_channel >>
+  CODACY_URL: << parameters.codacy_url >>
+steps:
+  - run:
+      name: Checkout codacy chart
+      command: |
+        git clone https://github.com/codacy/chart.git
+  - run:
+      name: Patch artifact version and helm repo
+      command: |
+        pip3 install yq
+        DEPENDENCIES=$(yq .dependencies chart/codacy/requirements.yaml | jq .[].name |sed "s/\"//g" | grep -v "minio\|rabbitmq-ha\|postgresql\|log-router")
```

Comment:
can't we just use dymanic versions on requirements.yaml for this and make use of helm tools to do this?

the `helm update .` and `requirements.lock` are exacly for this purpose

Reply:
also  `grep -v "minio\|rabbitmq-ha\|postgresql\|log-router"` will create a lot of troubles when we decide to add new components. Again, the requirements.yaml should be used to say which versions are dynamic and which are not.

Reply:
I have just addressed the grep part. 
We use jq to filter(select) only the components which have a git key in the requirements.yaml file. Those are the ones that belong to codacy and thus should be updated to the latest version.

Reply:
The problem with dynamic versions is that some of our components publish as SNAPSHOTS, while others don't. 
For the dynamic version to pattern match against a SNAPSHOT, you must define the version as
^5.3.1-SNAPSHOT for instance. 
^5.3.1 will not match any snapshot :-(

Reply:
this definitely looks like an hammer, I would rather go in the direction of uniforming the deployment of the components other than hammering here